pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java

```diff
@@ -926,8 +926,10 @@ private void deleteSegmentsInternal(String tableNameWithType, List<String> segme
   public String getServerMetadata(
       @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName,
       @ApiParam(value = "OFFLINE|REALTIME") @QueryParam("type") String tableTypeStr,
-      @ApiParam(value = "Columns name", allowMultiple = true) @QueryParam("columns") @DefaultValue("")
-          List<String> columns, @Context HttpHeaders headers) {
+      @Encoded @ApiParam(value = "Segments to include", allowMultiple = true) @QueryParam("segments")
+      @Nullable List<String> segments,
+      @Encoded @ApiParam(value = "Columns name", allowMultiple = true) @QueryParam("columns")
+      @Nullable List<String> columns, @Context HttpHeaders headers) {
     tableName = DatabaseUtils.translateTableName(tableName, headers);
     LOGGER.info("Received a request to fetch metadata for all segments for table {}", tableName);
```

Comment:
nit: should this be updated to call out "all segments" only if segments is empty? and otherwise log the number of segments?